src/hotspot/share/opto/superword.cpp

```diff
@@ -3718,7 +3718,10 @@ bool SuperWord::construct_bb() {
             // First see if we can map the reduction on the given system we are on, then
             // make a data entry operation for each reduction we see.
             BasicType bt = use->bottom_type()->basic_type();
-            if (ReductionNode::implemented(use->Opcode(), Matcher::min_vector_size(bt), bt)) {
+            // Matcher::min_vector_size may return 1 in some cases, e.g. double for x86.
+            // For vector reduction implemented check we need atleast two elements.
+            int min_vec_size = MAX2(Matcher::min_vector_size(bt), 2);
+            if (ReductionNode::implemented(use->Opcode(), min_vec_size, bt)) {
```

Comment:
Hi @sviswa7, can we use `superword_max_vector_size()` as the input here? `MAX2(Matcher::min_vector_size(bt), 2);` may not tally with the actual situation on other 64-bit platforms. WDYT?

Reply:
@fg1417 Thanks a lot for the review. Yes, we could use superword_max_vector_size here. I have made the change.